client/src/models/Location.js

```diff
@@ -177,4 +178,20 @@ export default class Location extends Model {
     }
     return this.name
   }
+
+  static LOCATION_CLIENT_SIDE_FIELDS = [
+    "displayedCoordinate",
+    DEFAULT_CUSTOM_FIELDS_PARENT
+  ]
+
+  static getObjClientSideFieldsFiltered(obj) {
```

Comment:
Perhaps this needs an optional second argument (or a variable argument list) for fields that you may want to remove incidentally (i.e. only in some places, not everywhere). Could then also become a generic function in `Model.js`, to be used by all models.